csharp/src/Ice/Connection.cs

```diff
@@ -107,23 +107,20 @@ public interface Connection
         ObjectPrx createProxy(Identity id);
 
         /// <summary>
-        /// Explicitly set an object adapter that dispatches requests that are received over this connection.
-        /// A client can
-        /// invoke an operation on a server using a proxy, and then set an object adapter for the outgoing connection that
-        /// is used by the proxy in order to receive callbacks. This is useful if the server cannot establish a connection
-        /// back to the client, for example because of firewalls.
+        /// Associates an object adapter with this connection. When a connection receives a request, it dispatches this
+        /// request using its associated object adapter. If the associated object adapter is null, the connection
+        /// rejects any incoming request with an <see cref="ObjectNotExistException" />.
+        /// The default object adapter of an incoming connection is the object adapter that created this connection;
```

Comment:
It is possible but unusual to call setAdapter on an incoming connection. This changes the dispatcher and OA used for the actual dispatch. This does not change the thread pool used for the dispatches:

https://github.com/zeroc-ice/ice/blob/b52ef66df58062faaa81c248e0bac0669cc989ee/csharp/src/Ice/ConnectionI.cs#L572

https://github.com/zeroc-ice/ice/blob/b52ef66df58062faaa81c248e0bac0669cc989ee/csharp/src/Ice/ConnectionI.cs#L1448

I find this behavior unexpected and it's not clear to me why we don't restrict setAdapter to outgoing connections.